File: VB/WpfApplication241/MainViewModel.vb
Imports System
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm

Namespace WpfApplication241

    Public Class MainViewModel
        Inherits ViewModelBase

        Public Property DisabledDates As ObservableCollection(Of Date)
            Get
                Return GetValue(Of ObservableCollection(Of Date))()
            End Get

            Set(ByVal value As ObservableCollection(Of Date))
                SetValue(value)
            End Set
        End Property

        Public Property MaxDate As Date
            Get
                Return GetValue(Of Date)()
            End Get

            Set(ByVal value As Date)
                SetValue(value)
            End Set
        End Property

        Public Property MinDate As Date
            Get
                Return GetValue(Of Date)()
            End Get

            Set(ByVal value As Date)
                SetValue(value)
            End Set
        End Property

        Public Sub New()
            MaxDate = Date.Today
            MinDate = MaxDate - New TimeSpan(90, 0, 0, 0)
            DisabledDates = New ObservableCollection(Of Date)() From {Date.Today.AddDays(1), Date.Today.AddDays(2), Date.Today.AddDays(3), Date.Today.AddDays(4)}
        End Sub
    End Class
End Namespace

File: CS/WpfApplication241/MainViewModel.cs
using System;
using System.Collections.ObjectModel;
using DevExpress.Mvvm;

namespace WpfApplication241 {
    public class MainViewModel : ViewModelBase {
        public ObservableCollection<DateTime> DisabledDates {
            get { return GetValue<ObservableCollection<DateTime>>(); }
            set { SetValue(value); }
        }
        public DateTime MaxDate {
            get { return GetValue<DateTime>(); }
            set { SetValue(value); }
        }
        public DateTime MinDate {
            get { return GetValue<DateTime>(); }
            set { SetValue(value); }
        }
        public MainViewModel() {
            MaxDate = DateTime.Today;
            MinDate = MaxDate - new TimeSpan(90, 0, 0, 0);
            DisabledDates = new ObservableCollection<DateTime>() {
                DateTime.Today.AddDays(1),
                DateTime.Today.AddDays(2),
                DateTime.Today.AddDays(3),
                DateTime.Today.AddDays(4),
            };
        }
    }
}


